contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -804,20 +927,61 @@ contract VaultHub is PausableUntilWithRoles {
             revert NoLiabilitySharesShouldBeLeft(_vault, liabilityShares_);
         }
 
+        // try to settle using existing unsettled fees, and revert if some obligations remain
+        _checkAndSettleObligations(_vault, _record, _connection, 0);
         _connection.pendingDisconnect = true;
     }
 
+    function _applyVaultReport(
+        address _vault,
+        VaultConnection storage _connection,
+        VaultRecord storage _record,
+        uint64 _reportTimestamp,
+        uint256 _reportTotalValue,
+        uint256 _reportLiabilityShares,
+        int256 _reportInOutDelta
+    ) internal {
+        uint256 liabilityShares_ = Math256.max(_record.liabilityShares, _reportLiabilityShares);
+        uint256 liability = _getPooledEthBySharesRoundUp(liabilityShares_);
+
+        uint256 lockedEther = Math256.max(
+            liability * TOTAL_BASIS_POINTS / (TOTAL_BASIS_POINTS - _connection.reserveRatioBP),
+            CONNECT_DEPOSIT
+        );
+
+        _record.locked = uint128(lockedEther);
+        _record.reportTimestamp = _reportTimestamp;
+        _record.report = Report({
+            totalValue: uint128(_reportTotalValue),
+            inOutDelta: int128(_reportInOutDelta)
+        });
+
+        IStakingVault vault_ = IStakingVault(_vault);
+        if (!_isVaultHealthy(_connection, _record) && !vault_.beaconChainDepositsPaused()) {
+            vault_.pauseBeaconChainDeposits();
```

Comment:
:warning: it's possible to maliciously lift this pause by backrunning settlement of obligations